src/index.ts

```diff
@@ -24,6 +24,18 @@ export const plugin: PluginFunction<{}> = async schema => {
         return `${name}: [${impls.map(n => `"${n}"`).join(", ")}],`;
       })}
     };
+    
+    ${addUnionTypes(interfaceImpls)}
   `.toStringWithImports();
+
+  // Also add type unions of the possible types for use in code if desired
+
+
   return { content } as PluginOutput;
 };
+
+function addUnionTypes(interfaceImpls: Record<string, string[]>) {
+  return Object.entries(interfaceImpls).map(([name, impls]) => `
+  export type ${name}Types = ${impls.join(" | ")};
```

Comment:
Really minor, maybe indent this `export` another two spaces so it indents past the `return`. So that it looks like its in the "right place" for reviewing this code generation code, and then prettier will format the output anyway on the way out.

Reply:
Ah, looks like I didn't commit the prettier changes. This looks a bit better now.